### ISSUE TITLE
[BUG:CLI] Vercel CLI does not support git worktrees

### ISSUE THREAD

**Issue by Ziothh** [zeit/now#11876]
When running most of the `vercel` cli commands I get this error:

I'm currently working in the worktree `WORKTREE_ROOT_DIR/fix/sso`

```
$ vercel dev
Error: ENOTDIR: not a directory, lstat '/Users/user/my-project/monorepo/fix/sso/.git/config'
```

Vercel CLI version:
```
$ vercel -v
Vercel CLI 34.1.2
34.1.2
```

**Comment by jmikedupont2** [comment]
yes same with submodules

**Comment by coagmano** [comment]
Was supposed to have been fixed by https://github.com/vercel/vercel/pull/11283 in vercel@33.7.0

Still have this issue with Vercel CLI 34.2.6

**Comment by hireAlanAyala** [comment]
I am having the same issue still

**Comment by Ziothh** [comment]
Error still occurs:
```bash
~/monorepo/branch $ vercel env pull

Vercel CLI 39.2.2
Error: ENOTDIR: not a directory, lstat '~/monorepo/branch/.git/config'
```

**Comment by anaisbetts** [comment]
This bug really hurts in the vibecoding era where you _really_ want to be able to create subtrees for new agents and call `vercel pull`

**Comment by financialvice** [comment]
my intuition is that there would be some `vercel.json` config for this, but it seems like there is not

in the same boat as @anaisbetts , git worktree support has only recently become important for me but it is very important and becoming a huge part of my workflow

**Comment by rutmehta** [comment]
> This bug really hurts in the vibecoding era where you _really_ want to be able to create subtrees for new agents and call `vercel pull`

still an issue

**Comment by royteeuwen** [comment]
Yep, still an issue and very annoying. Any workarounds available?

**Comment by louiskhub** [comment]
I think #14867 fixed this! Confirmed that it's working in `vercel@50.13.2`

**Comment by rutmehta** [comment]
> I think [#14867](https://github.com/vercel/vercel/pull/14867) fixed this! Confirmed that it's working in `vercel@50.13.2`

still isn't working for me
